lib/mapper.js

```diff
@@ -150,14 +150,24 @@ exports.orderCompleted = function(track, settings){
     };
 
     productPayload.properties = {
-      $event_id: product.productId() || product.id() || track.orderId() + '_' + product.sku(),
       $value: product.price(),
       Name: product.name(),
       Quantity: product.quantity(),
       "Product Categories": [product.category()],
       SKU: product.sku()
     };
 
+    // ensure unique $event_id is associated with each Ordered Product event by combining Order Completed order_id and product's productId or SKU
+    var orderId = payloads.order.properties.$event_id;
```

Comment:
Ahh yeah then I'd declare the variable `orderId` outside of the scope and then reference the variable where needed all throughout @brennan